src/python/pants/core/util_rules/external_tool.py

```diff
@@ -175,6 +224,35 @@ def get_request(self, plat: Platform) -> ExternalToolRequest:
             f"{self.known_versions}"
         )
 
+    @classmethod
+    def check_version_constraints(
+        cls, version: str, constraints: Requirement, action: UnsupportedVersionUsage
+    ) -> None:
+        if constraints.specifier.contains(version):  # type: ignore[attr-defined]
+            # all ok
+            return None
+
+        msg = [
+            f"The option [{cls.options_scope}].version is set to {version}, which is not compatible with what this release of Pants expects: {constraints}",
+        ]
+
+        if action is UnsupportedVersionUsage.Allow:
+            msg.append(
+                f"The option [{cls.options_scope}].use_unsupported_version is set to 'allow'."
+            )
+            logger.info("\n".join(msg))
```

Comment:
I think you were probably right for this to be `debug` previously. Otherwise, it's nosiy to get this log every time when you said you want to ignore issues.

But imo, even better is to get rid of the `allow` option and instead suggest in the warning: 

> Alternatively, you can ignore this warning (at your own peril) by adding this to the `GLOBAL` section of `pants.toml`: `ignore_warnings = ["The option [foo].version is set to"]`.

Reply:
Yeah, I too was leaning towards removing `allow` as well..  but wasn't aware of the `ignore_warnings` option :D

Reply:
Great! Please do double check that your suggested pants.toml entry will work - it's an exact string match so needs to be done exactly.

Reply:
It did work! neat 👍

Reply:
```
$ ./pants --download-pex-bin-use-unsupported-version=warning --changed-since=HEAD fmt
22:47:43.57 [WARN] The option [download-pex-bin].version is set to v2.1.42, which is not compatible with what this release of Pants expects: pex<3.0,>=2.1.43. Please update the version to a supported value, or consider using a different Pants release if you cannot change the version. Alternatively, you can ignore this warning (at your own peril) by adding this to the GLOBAL section of pants.toml: ignore_warnings = ["The option [download-pex-bin].version is set to"].
22:47:46.02 [INFO] Initializing scheduler...
```

that warning went away by copy pasting the suggested config to `pants.toml` ;)

Reply:
(temp changed the constraint to trigger the warning)